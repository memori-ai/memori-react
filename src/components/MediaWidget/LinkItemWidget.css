.memori-link-items--grid {
    display: grid;
    padding: 1rem;
    grid-auto-flow: dense;
    grid-gap: 1rem;
    grid-template-columns: repeat(auto-fill, minmax(150px, 1fr));
}

.memori-link-item--card {
    height: 100%;
    margin-bottom: 1rem;
}

.memori-link-item--card .memori-card--cover {
    height: auto;
    text-align: center;
}

.memori-link-item--card .memori-card--cover img {
    min-width: 100%;
    min-height: 100%;
    object-fit: contain;
}

a.memori-link-item--link {
    text-decoration: none;
}

.memori-link-item--card-cover-icon {
    padding: 1rem;
}

.memori-link-item--icon {
    width: 50%;
    height: 50%;
}