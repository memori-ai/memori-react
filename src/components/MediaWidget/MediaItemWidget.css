.memori-media-items--grid {
    display: grid;
    padding: 1rem;
    grid-auto-flow: dense;
    grid-gap: 1rem;
    grid-template-columns: repeat(auto-fill, minmax(150px, 1fr));
}

.memori-media-item--card {
    margin-bottom: 1rem;
}

.memori-media-item--image .memori-card--content {
    padding: 0;
}

.memori-media-item .memori-card--cover {
    height: auto;
}

.memori-media-item .memori-card--cover audio,
.memori-media-item .memori-card--cover video {
    border-radius: 5px;
}

.memori-media-item--image .memori-card--cover {
    height: auto;
    background: transparent;
}

.memori-media-item--figure {
    display: flex;
    width: 100%;
    height: 100%;
    max-height: 150px;
    flex-direction: column;
    align-items: center;
    justify-content: center;
}

.memori-media-item--rgb-item {
    width: 100%;
    min-height: 100px;
    flex: 1;
}

.memori-media-item--figure img {
    min-width: 100%;
    min-height: 100%;
    object-fit: cover;
    object-position: center;
}

.memori-media-item--figure-caption {
    padding: 0.25rem 1rem;
    margin: 0.25rem 0;
    color: #666;
    font-size: 0.75rem;
    font-style: italic;
    font-weight: 400;
    text-align: center;
    word-break: break-word;
}

a.memori-media-item--link {
    text-decoration: none;
}

.memori-media-item--icon {
    width: 50%;
    height: 50%;
}

.memori-media-item--modal .memori-media-item--figure {
    position: relative;
    z-index: 0;
    width: 100%;
    max-height: 100%;
    flex-direction: row;
}

.memori-media-item--modal .memori-media-item--figure img {
    object-fit: contain;
}

.memori-media-item--modal .memori-card {
    width: 100%;
    max-width: 100%;
    height: 100%;
    max-height: 100%;
    border-radius: 0;
    margin-bottom: 0;
    box-shadow: none;
}

.memori-media-item--modal .memori-card--content {
    display: none;
}

.memori-media-item--modal .memori-card--cover {
    height: 100%;
    border-radius: 0;
}

.memori-media-item--modal .memori-modal--panel {
    width: 75vw;
    height: 85vh;
    -webkit-backdrop-filter: blur(var(--memori-blur-background, 10px));
    backdrop-filter: blur(var(--memori-blur-background, 10px));
    background: rgba(0, 0, 0, 0.25);
}

.memori-media-item--modal .memori-modal--panel .memori-card {
    background: none;
}

.memori-media-item--modal .memori-modal--panel .memori-spin {
    display: flex;
    height: 100%;
    align-items: center;
    justify-content: center;
}

.memori-media-item--modal a.memori-media-item--link {
    display: flex;
    width: 100%;
    height: 100%;
    align-items: center;
    justify-content: center;
}

.memori-media-item--modal .memori-media-item--figure-caption {
    position: absolute;
    bottom: 0;
    width: 100%;
    -webkit-backdrop-filter: blur(var(--memori-blur-background, 5px));
    backdrop-filter: blur(var(--memori-blur-background, 5px));
    background: rgba(255, 255, 255, 0.5);
    color: #000;
}

.memori-media-item--modal .memori-modal--close button {
    border-color: #fff;
    color: #fff;
}
