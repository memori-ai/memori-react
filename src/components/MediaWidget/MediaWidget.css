.actionButton {
    display: flex;
    align-items: center;
    padding: 10px 30px;
    border-radius: 5px;
    background-color: var(--memori-primary);
    color: var(--memori-primary-text);
    transition: all 0.5s ease;
}

.actionButton:hover {
    border-color: var(--memori-primary) !important;
    background-color: var(--memori-primary);
    color: var(--light-gray) !important;
    filter: brightness(1.08);
}

.actionButton:disabled {
    background-color: lightgray !important;
}

.memori-media--hints {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    padding-left: 0;
    margin-top: 2rem;
    list-style: none;
}

.memori-media--hints li {
    margin-bottom: 0.5rem;
}

.memori-media--hints li + li {
    margin-left: 0.5rem;
}

.ease-out {
    transition-timing-function: ease-out;
}

.ease-in {
    transition-timing-function: ease-in;
}

.duration-500 {
    transition-duration: 500ms;
}

.duration-300 {
    transition-duration: 300ms;
}

.duration-200 {
    transition-duration: 200ms;
}

.opacity-0 {
    opacity: 0;
}

.opacity-100 {
    opacity: 1;
}

.scale-95 {
    transform: scale(0.95);
}

.scale-100 {
    transform: scale(1);
}

.translate-y-1 {
    transform: translateY(1em);
}

.translate-y-0 {
    transform: translateY(0);
}

.delay-100 {
    transition-delay: 100ms;
}

.delay-200 {
    transition-delay: 200ms;
}

.delay-300 {
    transition-delay: 300ms;
}

.delay-400 {
    transition-delay: 400ms;
}

.delay-500 {
    transition-delay: 500ms;
}

.delay-600 {
    transition-delay: 600ms;
}

.delay-700 {
    transition-delay: 700ms;
}

.delay-800 {
    transition-delay: 800ms;
}

.delay-900 {
    transition-delay: 900ms;
}

.delay-1000 {
    transition-delay: 1000ms;
}

.memori-hints--show-more {
    text-align: center;
}

button.memori-hints--show-more-button {
    border-radius: 5px;
    background: #fff;
    color: #333;
}
