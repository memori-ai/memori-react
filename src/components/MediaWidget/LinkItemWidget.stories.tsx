import React from 'react';
import { Meta, Story } from '@storybook/react';
import I18nWrapper from '../../I18nWrapper';
import LinkItemWidget, { Props } from './LinkItemWidget';

import './LinkItemWidget.css';

const meta: Meta = {
  title: 'Media Widget/Link',
  component: LinkItemWidget,
  argTypes: {},
  parameters: {
    controls: { expanded: true },
  },
};

export default meta;

const Template: Story<Props> = args => (
  <I18nWrapper>
    <LinkItemWidget {...args} />
  </I18nWrapper>
);

// By passing using the Args format for exported stories, you can control the props for a component for reuse in a test
// https://storybook.js.org/docs/react/workflows/unit-testing
export const Default = Template.bind({});
Default.args = {
  items: [
    {
      mediumID: '95226d7e-7bae-465e-8b80-995587bb5971',
      mimeType: 'text/html',
      title: 'Memori Srl',
      url: 'https://memori.ai/en',
    },
    {
      mediumID: '95226d7e-7bae-465e-8b80-995587bb5971',
      mimeType: 'text/html',
      title: 'AIsuru',
      url: 'https://www.aisuru.com/en',
    },
    {
      mediumID: '95226d7e-7bae-465e-8b80-995587bb5971',
      mimeType: 'text/html',
      title: 'Introducing Plone Remix | Vimeo',
      url: 'https://vimeo.com/766468314',
    },
    {
      mediumID: '95226d7e-7bae-465e-8b80-995587bb5971',
      mimeType: 'text/html',
      title: 'A sustainable web: is it possible? - Nicola Zambello | YouTube',
      url: 'https://www.youtube.com/watch?v=feH26j3rBz8',
    },
    {
      mediumID: '95226d7e-7bae-465e-8b80-995587bb5971',
      mimeType: 'text/html',
      title: 'Memori backend API',
      url: 'https://backend.memori.ai',
    },
  ],
};
