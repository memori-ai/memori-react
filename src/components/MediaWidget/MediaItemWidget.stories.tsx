import React from 'react';
import { Meta, Story } from '@storybook/react';
import I18nWrapper from '../../I18nWrapper';
import MediaItemWidget, { Props } from './MediaItemWidget';

import './MediaItemWidget.css';

const meta: Meta = {
  title: 'Media Widget/Media',
  component: MediaItemWidget,
  argTypes: {
    translateTo: {
      control: {
        type: 'select',
        items: ['IT', 'EN', 'DE', 'FR', 'ES', 'PT', 'RU', 'ZH'],
      },
    },
  },
  parameters: {
    controls: { expanded: true },
  },
};

export default meta;

const Template: Story<Props> = args => (
  <I18nWrapper>
    <MediaItemWidget {...args} />
  </I18nWrapper>
);

// By passing using the Args format for exported stories, you can control the props for a component for reuse in a test
// https://storybook.js.org/docs/react/workflows/unit-testing
export const Default = Template.bind({});
Default.args = {
  items: [
    {
      mediumID: '95226d7e-7bae-465e-8b80-995587bb5971',
      mimeType: 'text/javascript',
      title: 'Snippet',
      content: "console.log('Hello World!');",
    },
    {
      mediumID: '95226d7e-7bae-465e-8b80-995587bb5972',
      mimeType: 'image/png',
      title: 'Image',
      url: 'https://picsum.photos/200/300',
      content: 'https://picsum.photos/200/300',
    },
    {
      mediumID: '95226d7e-7bae-465e-8b80-995587bb5973',
      mimeType: 'application/pdf',
      title: 'PDF',
      url: 'https://www.w3.org/WAI/ER/tests/xhtml/testfiles/resources/pdf/dummy.pdf',
      content:
        'https://www.w3.org/WAI/ER/tests/xhtml/testfiles/resources/pdf/dummy.pdf',
    },
    {
      mediumID: '95226d7e-7bae-465e-8b80-995587bb5974',
      mimeType: 'video/mp4',
      title: 'Video',
      url: 'https://www.w3schools.com/html/mov_bbb.mp4',
      content: 'https://www.w3schools.com/html/mov_bbb.mp4',
    },
    {
      mediumID: '95226d7e-7bae-465e-8b80-995587bb5975',
      mimeType: 'audio/mpeg',
      title: 'Audio',
      url: 'https://www.w3schools.com/html/horse.mp3',
      content: 'https://www.w3schools.com/html/horse.mp3',
    },
    {
      mediumID: '95226d7e-7bae-465e-8b80-995587bb5976',
      mimeType: 'application/msword',
      title: 'Word',
      url: 'https://www.w3.org/WAI/ER/tests/xhtml/testfiles/resources/pdf/dummy.pdf',
      content:
        'https://www.w3.org/WAI/ER/tests/xhtml/testfiles/resources/pdf/dummy.pdf',
    },
    {
      mediumID: '95226d7e-7bae-465e-8b80-995587bb5977',
      mimeType: 'application/vnd.ms-excel',
      title: 'Excel',
      url: 'https://www.w3.org/WAI/ER/tests/xhtml/testfiles/resources/pdf/dummy.pdf',
      content:
        'https://www.w3.org/WAI/ER/tests/xhtml/testfiles/resources/pdf/dummy.pdf',
    },
    {
      mediumID: '65ca4a6d-f20b-402e-9d79-5e470f247928',
      mimeType: 'image/png',
      title: 'Verde',
      url: 'rgb(0, 255, 0)',
    },
  ],
};

export const ImagesGrid = Template.bind({});
ImagesGrid.args = {
  items: Array.from({ length: 9 }, (_, i) => ({
    mediumID: `95226d7e-7bae-465e-8b80-995587bb597${i}`,
    mimeType: 'image/png',
    title: `Image ${i}`,
    url: `https://picsum.photos/${i % 2 ? '200' : '300'}/${
      i % 3 ? '300' : '200'
    }?random=${i}`,
  })),
};

export const WithCustomMediaRenderer = Template.bind({});
WithCustomMediaRenderer.args = {
  items: [
    {
      mediumID: '95226d7e-7bae-465e-8b80-995587bb5971',
      mimeType: 'image/png',
      title: 'Image',
      url: 'https://picsum.photos/200/300',
    },
    {
      mediumID: '95226d7e-7bae-465e-8b80-995587bb5973',
      mimeType: 'application/pdf',
      title: 'PDF',
      url: 'https://www.w3.org/WAI/ER/tests/xhtml/testfiles/resources/pdf/dummy.pdf',
      content:
        'https://www.w3.org/WAI/ER/tests/xhtml/testfiles/resources/pdf/dummy.pdf',
    },
  ],
  customMediaRenderer: (mimeType: string) => {
    if (mimeType === 'image/png') {
      return <div>Custom Image Renderer</div>;
    }
  },
};

export const WithRGBColors = Template.bind({});
WithRGBColors.args = {
  items: [
    {
      mediumID: '65ca4a6d-f20b-402e-9d79-5e470f247927',
      mimeType: 'image/png',
      title: 'Rosso',
      url: 'rgb(255, 0, 0)',
    },
    {
      mediumID: '65ca4a6d-f20b-402e-9d79-5e470f247928',
      mimeType: 'image/png',
      title: 'Verde',
      url: 'rgb(0, 255, 0)',
    },
    {
      mediumID: '65ca4a6d-f20b-402e-9d79-5e470f247929',
      mimeType: 'image/png',
      title: 'Blu',
      url: 'rgb(0, 0, 255)',
    },
  ],
};
