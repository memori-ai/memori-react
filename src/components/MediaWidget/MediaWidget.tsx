import {
  Medium,
  TranslatedHint,
} from '@memori.ai/memori-api-client/dist/types';
import React, { useEffect, useState, memo } from 'react';
import Button from '../ui/Button';
import LinkItemWidget from './LinkItemWidget';
import MediaItemWidget, { Props as MediaItemProps } from './MediaItemWidget';
import { Transition } from '@headlessui/react';
import cx from 'classnames';
import { useTranslation } from 'react-i18next';

export interface Props {
  hints?: TranslatedHint[];
  links?: Medium[];
  media?: Medium[];
  simulateUserPrompt?: (item: string, translatedItem?: string) => void;
  sessionID?: string;
  baseUrl?: string;
  apiUrl?: string;
  translateTo?: string;
  customMediaRenderer?: MediaItemProps['customMediaRenderer'];
}

const MediaWidget: React.FC<Props> = ({
  hints = [],
  links = [],
  media = [],
  simulateUserPrompt = () => {},
  sessionID,
  baseUrl,
  apiUrl,
  translateTo,
  customMediaRenderer,
}: Props) => {
  const { t } = useTranslation();
  const [showHints, setShowHints] = useState(true);
  const [hintsPagination, setHintsPagination] = useState(6);

  useEffect(() => {
    setShowHints(true);
  }, [hints]);

  return (
    <div className="memori-media-widget">
      {media?.length > 0 && (
        <MediaItemWidget
          items={media}
          sessionID={sessionID}
          translateTo={translateTo}
          baseURL={baseUrl}
          apiURL={apiUrl}
          customMediaRenderer={customMediaRenderer}
        />
      )}
      {links?.length > 0 && <LinkItemWidget items={links} baseUrl={baseUrl} />}
      {hints?.length > 0 && showHints && (
        <>
          <Transition appear show as="ul" className="memori-media--hints">
            {hints.slice(0, hintsPagination).map((item, index) => (
              <Transition.Child
                as="li"
                key={item.text + index}
                enter="ease-out duration-500"
                enterFrom="opacity-0 translate-y-1"
                enterTo="opacity-1 translate-y-0"
                leave="ease-in duration-300"
                leaveFrom="opacity-1"
                leaveTo="opacity-0"
              >
                <Button
                  key={item.text + index}
                  className={cx('memori-media--hint')}
                  primary
                  onClick={() => {
                    simulateUserPrompt(item.originalText, item.text);
                    setShowHints(false);
                  }}
                  onTouchEnd={() => {
                    simulateUserPrompt(item.originalText, item.text);
                    setShowHints(false);
                  }}
                >
                  {item.text}
                </Button>
              </Transition.Child>
            ))}
          </Transition>
          {hints.length > hintsPagination && (
            <div className="memori-hints--show-more">
              <Button
                className="memori-hints--show-more-button"
                id="showMoreHints"
                onClick={() => setHintsPagination(hintsPagination + 6)}
              >
                {t('expand') || 'Expand'}
              </Button>
            </div>
          )}
        </>
      )}
    </div>
  );
};

export default memo(MediaWidget);
