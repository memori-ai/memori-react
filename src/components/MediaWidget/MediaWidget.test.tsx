import React from 'react';
import {
  Medium,
  TranslatedHint,
} from '@memori.ai/memori-api-client/dist/types';
import { render } from '@testing-library/react';
import { medium } from '../../mocks/data';
import MediaWidget from './MediaWidget';

const linkMedium: Medium = {
  mediumID: '95226d7e-7bae-465e-8b80-995587bb5971',
  mimeType: 'text/html',
  title: 'Memori Srl',
  url: 'https://memori.ai/en',
};

const hint: TranslatedHint = {
  text: 'All right',
  originalText: 'Va bene',
};

it('renders MediaWidget as empty unchanged', () => {
  const { container } = render(<MediaWidget />);
  expect(container).toMatchSnapshot();
});

it('renders MediaWidget with links unchanged', () => {
  const { container } = render(<MediaWidget links={[linkMedium]} />);
  expect(container).toMatchSnapshot();
});

it('renders MediaWidget with media unchanged', () => {
  const { container } = render(<MediaWidget media={[medium]} />);
  expect(container).toMatchSnapshot();
});

it('renders MediaWidget with hints unchanged', () => {
  const { container } = render(<MediaWidget hints={[hint]} />);
  expect(container).toMatchSnapshot();
});
