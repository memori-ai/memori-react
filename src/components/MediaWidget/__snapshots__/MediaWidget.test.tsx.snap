// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MediaWidget as empty unchanged 1`] = `
<div>
  <div
    class="memori-media-widget"
  />
</div>
`;

exports[`renders MediaWidget with hints unchanged 1`] = `
<div>
  <div
    class="memori-media-widget"
  >
    <ul
      class="memori-media--hints"
    >
      <li
        class="ease-out duration-500 opacity-0 translate-y-1"
      >
        <button
          class="memori-button memori-button--primary memori-button--rounded memori-button--padded memori-media--hint"
        >
          All right
        </button>
      </li>
    </ul>
  </div>
</div>
`;

exports[`renders MediaWidget with links unchanged 1`] = `
<div>
  <div
    class="memori-media-widget"
  >
    <div
      class="memori-link-items"
    >
      <div
        class="memori-link-items--grid"
      >
        <div
          class="memori-link-item ease-out duration-500 delay-0 opacity-0 scale-95"
        >
          <a
            class="memori-link-item--link"
            href="https://memori.ai/en"
            rel="noopener noreferrer"
            target="_blank"
          >
            <div
              class="memori-card memori-link-item--card memori-card--with-cover memori-card--hoverable"
            >
              <div
                class="memori-spin"
              >
                <div
                  class="memori-card--cover"
                >
                  <div
                    class="memori-link-item--card-cover-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="memori-link-item--icon"
                      focusable="false"
                      role="img"
                      viewBox="0 0 1024 1024"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M574 665.4a8.03 8.03 0 0 0-11.3 0L446.5 781.6c-53.8 53.8-144.6 59.5-204 0-59.5-59.5-53.8-150.2 0-204l116.2-116.2c3.1-3.1 3.1-8.2 0-11.3l-39.8-39.8a8.03 8.03 0 0 0-11.3 0L191.4 526.5c-84.6 84.6-84.6 221.5 0 306s221.5 84.6 306 0l116.2-116.2c3.1-3.1 3.1-8.2 0-11.3L574 665.4zm258.6-474c-84.6-84.6-221.5-84.6-306 0L410.3 307.6a8.03 8.03 0 0 0 0 11.3l39.7 39.7c3.1 3.1 8.2 3.1 11.3 0l116.2-116.2c53.8-53.8 144.6-59.5 204 0 59.5 59.5 53.8 150.2 0 204L665.3 562.6a8.03 8.03 0 0 0 0 11.3l39.8 39.8c3.1 3.1 8.2 3.1 11.3 0l116.2-116.2c84.5-84.6 84.5-221.5 0-306.1zM610.1 372.3a8.03 8.03 0 0 0-11.3 0L372.3 598.7a8.03 8.03 0 0 0 0 11.3l39.6 39.6c3.1 3.1 8.2 3.1 11.3 0l226.4-226.4c3.1-3.1 3.1-8.2 0-11.3l-39.5-39.6z"
                      />
                    </svg>
                  </div>
                </div>
                <div
                  class="memori-card--content"
                >
                  <h3
                    class="memori-card--title"
                  >
                    Memori Srl
                  </h3>
                  <div
                    class="memori-card--children"
                  />
                </div>
                <div
                  class="memori-spin--spinner"
                >
                  <svg
                    aria-hidden="true"
                    class="memori-loading-icon"
                    focusable="false"
                    role="img"
                    viewBox="0 0 1024 1024"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders MediaWidget with media unchanged 1`] = `
<div>
  <div
    class="memori-media-widget"
  >
    <div
      class="memori-media-items"
    >
      <div
        class="memori-media-items--grid"
      >
        <div
          class="memori-media-item ease-out duration-500 delay-0 opacity-0 scale-95"
        >
          <a
            class="memori-media-item--link"
            href="https://api.lorem.space/image/game?w=150&h=220&hash=8B7BCDC2"
            rel="noopener noreferrer"
            target="_blank"
            title="Game Cover"
          >
            <div
              class="memori-card memori-media-item--card memori-media-item--image memori-card--with-cover memori-card--hoverable"
            >
              <div
                class="memori-spin"
              >
                <div
                  class="memori-card--cover"
                >
                  <picture
                    class="memori-media-item--figure"
                  >
                    <source
                      srcset="https://api.lorem.space/image/game?w=150&h=220&hash=8B7BCDC2, https://api.lorem.space/image/game?w=150&h=220&hash=8B7BCDC2, "
                      type="image/jpeg"
                    />
                    <img
                      alt="Game Cover"
                      src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAMIAAADDCAYAAADQvc6UAAABRWlDQ1BJQ0MgUHJvZmlsZQAAKJFjYGASSSwoyGFhYGDIzSspCnJ3UoiIjFJgf8LAwSDCIMogwMCcmFxc4BgQ4ANUwgCjUcG3awyMIPqyLsis7PPOq3QdDFcvjV3jOD1boQVTPQrgSkktTgbSf4A4LbmgqISBgTEFyFYuLykAsTuAbJEioKOA7DkgdjqEvQHEToKwj4DVhAQ5A9k3gGyB5IxEoBmML4BsnSQk8XQkNtReEOBxcfXxUQg1Mjc0dyHgXNJBSWpFCYh2zi+oLMpMzyhRcASGUqqCZ16yno6CkYGRAQMDKMwhqj/fAIcloxgHQqxAjIHBEugw5sUIsSQpBobtQPdLciLEVJYzMPBHMDBsayhILEqEO4DxG0txmrERhM29nYGBddr//5/DGRjYNRkY/l7////39v///y4Dmn+LgeHANwDrkl1AuO+pmgAAADhlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAAqACAAQAAAABAAAAwqADAAQAAAABAAAAwwAAAAD9b/HnAAAHlklEQVR4Ae3dP3PTWBSGcbGzM6GCKqlIBRV0dHRJFarQ0eUT8LH4BnRU0NHR0UEFVdIlFRV7TzRksomPY8uykTk/zewQfKw/9znv4yvJynLv4uLiV2dBoDiBf4qP3/ARuCRABEFAoBEgghggQAQZQKAnYEaQBAQaASKIAQJEkAEEegJmBElAoBEgghggQAQZQKAnYEaQBAQaASKIAQJEkAEEegJmBElAoBEgghggQAQZQKAnYEaQBAQaASKIAQJEkAEEegJmBElAoBEgghggQAQZQKAnYEaQBAQaASKIAQJEkAEEegJmBElAoBEgghggQAQZQKAnYEaQBAQaASKIAQJEkAEEegJmBElAoBEgghggQAQZQKAnYEaQBAQaASKIAQJEkAEEegJmBElAoBEgghggQAQZQKAnYEaQBAQaASKIAQJEkAEEegJmBElAoBEgghggQAQZQKAnYEaQBAQaASKIAQJEkAEEegJmBElAoBEgghggQAQZQKAnYEaQBAQaASKIAQJEkAEEegJmBElAoBEgghggQAQZQKAnYEaQBAQaASKIAQJEkAEEegJmBElAoBEgghggQAQZQKAnYEaQBAQaASKIAQJEkAEEegJmBElAoBEgghggQAQZQKAnYEaQBAQaASKIAQJEkAEEegJmBElAoBEgghgg0Aj8i0JO4OzsrPv69Wv+hi2qPHr0qNvf39+iI97soRIh4f3z58/u7du3SXX7Xt7Z2enevHmzfQe+oSN2apSAPj09TSrb+XKI/f379+08+A0cNRE2ANkupk+ACNPvkSPcAAEibACyXUyfABGm3yNHuAECRNgAZLuYPgEirKlHu7u7XdyytGwHAd8jjNyng4OD7vnz51dbPT8/7z58+NB9+/bt6jU/TI+AGWHEnrx48eJ/EsSmHzx40L18+fLyzxF3ZVMjEyDCiEDjMYZZS5wiPXnyZFbJaxMhQIQRGzHvWR7XCyOCXsOmiDAi1HmPMMQjDpbpEiDCiL358eNHurW/5SnWdIBbXiDCiA38/Pnzrce2YyZ4//59F3ePLNMl4PbpiL2J0L979+7yDtHDhw8vtzzvdGnEXdvUigSIsCLAWavHp/+qM0BcXMd/q25n1vF57TYBp0a3mUzilePj4+7k5KSLb6gt6ydAhPUzXnoPR0dHl79WGTNCfBnn1uvSCJdegQhLI1vvCk+fPu2ePXt2tZOYEV6/fn31dz+shwAR1sP1cqvLntbEN9MxA9xcYjsxS1jWR4AIa2Ibzx0tc44fYX/16lV6NDFLXH+YL32jwiACRBiEbf5KcXoTIsQSpzXx4N28Ja4BQoK7rgXiydbHjx/P25TaQAJEGAguWy0+2Q8PD6/Ki4R8EVl+bzBOnZY95fq9rj9zAkTI2SxdidBHqG9+skdw43borCXO/ZcJdraPWdv22uIEiLA4q7nvvCug8WTqzQveOH26fodo7g6uFe/a17W3+nFBAkRYENRdb1vkkz1CH9cPsVy/jrhr27PqMYvENYNlHAIesRiBYwRy0V+8iXP8+/fvX11Mr7L7ECueb/r48eMqm7FuI2BGWDEG8cm+7G3NEOfmdcTQw4h9/55lhm7DekRYKQPZF2ArbXTAyu4kDYB2YxUzwg0gi/41ztHnfQG26HbGel/crVrm7tNY+/1btkOEAZ2M05r4FB7r9GbAIdxaZYrHdOsgJ/wCEQY0J74TmOKnbxxT9n3FgGGWWsVdowHtjt9Nnvf7yQM2aZU/TIAIAxrw6dOnAWtZZcoEnBpNuTuObWMEiLAx1HY0ZQJEmHJ3HNvGCBBhY6jtaMoEiJB0Z29vL6ls58vxPcO8/zfrdo5qvKO+d3Fx8Wu8zf1dW4p/cPzLly/dtv9Ts/EbcvGAHhHyfBIhZ6NSiIBTo0LNNtScABFyNiqFCBChULMNNSdAhJyNSiECRCjUbEPNCRAhZ6NSiAARCjXbUHMCRMjZqBQiQIRCzTbUnAARcjYqhQgQoVCzDTUnQIScjUohAkQo1GxDzQkQIWejUogAEQo121BzAkTI2agUIkCEQs021JwAEXI2KoUIEKFQsw01J0CEnI1KIQJEKNRsQ80JECFno1KIABEKNdtQcwJEyNmoFCJAhELNNtScABFyNiqFCBChULMNNSdAhJyNSiECRCjUbEPNCRAhZ6NSiAARCjXbUHMCRMjZqBQiQIRCzTbUnAARcjYqhQgQoVCzDTUnQIScjUohAkQo1GxDzQkQIWejUogAEQo121BzAkTI2agUIkCEQs021JwAEXI2KoUIEKFQsw01J0CEnI1KIQJEKNRsQ80JECFno1KIABEKNdtQcwJEyNmoFCJAhELNNtScABFyNiqFCBChULMNNSdAhJyNSiECRCjUbEPNCRAhZ6NSiAARCjXbUHMCRMjZqBQiQIRCzTbUnAARcjYqhQgQoVCzDTUnQIScjUohAkQo1GxDzQkQIWejUogAEQo121BzAkTI2agUIkCEQs021JwAEXI2KoUIEKFQsw01J0CEnI1KIQJEKNRsQ80JECFno1KIABEKNdtQcwJEyNmoFCJAhELNNtScABFyNiqFCBChULMNNSdAhJyNSiEC/wGgKKC4YMA4TAAAAABJRU5ErkJggg=="
                    />
                    <figcaption
                      class="memori-media-item--figure-caption"
                    >
                      Game Cover
                    </figcaption>
                  </picture>
                </div>
                <div
                  class="memori-card--content"
                >
                  <div
                    class="memori-card--children"
                  />
                </div>
                <div
                  class="memori-spin--spinner"
                >
                  <svg
                    aria-hidden="true"
                    class="memori-loading-icon"
                    focusable="false"
                    role="img"
                    viewBox="0 0 1024 1024"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
